### PR TITLE
Better response when invalid user deletes wrong subscription

### PR DESCRIPTION
**Pull request by a-breid-ericsson**
Fixing better response when a user tries to delete someone elses subscription. Maybe the gui should handle this before the request are made, but anyhow right now the message is better.

### Applicable Issues
A response was displayed for a user that the subscription did not exist if a user tried to delete a subscription made by someone else. But id does exist. Or at leas I can see it, maybe it is just in my mind...

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java [hunk 1]
```diff
@@ -56,6 +57,7 @@
 
     private static final String SUBSCRIPTION_NOT_FOUND = "Subscription is not found";
     private static final String SUBSCRIPTION_ALREADY_EXISTS = "Subscription already exists";
+    private static final String INVALID_USER = "Invalid user! You must be logged in as the creater to a subscription modify it.";
```
Comment (e-pettersson-ericsson) [inline src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java]: creater -> creator

Comment (e-pettersson-ericsson) [inline src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java]: Also, I would prefer if we add "Unauthorized" instead of invalid user.

Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java]: 'You must be logged in as the creator of a subscription to modify it'

Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java]: Done.

Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java]: Done.

File: src/main/java/com/ericsson/ei/services/SubscriptionService.java [hunk 2]
```diff
@@ -133,19 +134,25 @@ public boolean modifySubscription(Subscription subscription, String subscription
...
 
     @Override
-    public boolean deleteSubscription(String subscriptionName) {
-        String ldapUserName = (authenticate) ? HttpSessionConfig.getCurrentUser() : "";
-        String query = generateQuery(subscriptionName, ldapUserName);
-        boolean result = subscriptionRepository.deleteSubscription(query);
+    public boolean deleteSubscription(String subscriptionName) throws NotOwnerException {
+        String ldapUserName = (ldapEnabled) ? HttpSessionConfig.getCurrentUser() : "";
+        String deleteQuery = generateQuery(subscriptionName, ldapUserName);
+        String fetchQuery = generateQuery(subscriptionName, "");
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/services/SubscriptionService.java]: fetchquery seems not to be used, or?

File: src/main/java/com/ericsson/ei/services/SubscriptionService.java [hunk 3]
```diff
@@ -57,7 +58,7 @@
     private String repeatFlagHandlerCollection;
 
     @Value("${ldap.enabled}")
-    private boolean authenticate;
+    private boolean ldapEnabled;
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/services/SubscriptionService.java]: if you change name here, then you can change the name in all other places where the "authenticate" name is used. So we use same name in all places.

File: src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java [hunk 4]
```diff
@@ -170,11 +172,17 @@
...
+
+            try {
+                if (subscriptionService.deleteSubscription(subscriptionName)) {
+                    LOG.debug("Subscription is deleted successfully: " + subscriptionName);
+                } else {
+                    LOG.error("Subscription to delete is not found: " + subscriptionName);
+                    errorMap.put(subscriptionName, SUBSCRIPTION_NOT_FOUND);
+                }
+            } catch (NotOwnerException e) {
+                LOG.error("Unauthorized, wrong username: " + subscriptionName);
```
Comment (evasiba-ericsson) [inline src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java]: you log two error messages for the same exception. I thin it is better to remove the error logging here and keep the error logging where the exception is created.

Alternatively create an exception with the message you log when you create it and log it here.

File: src/main/java/com/ericsson/ei/services/SubscriptionService.java (hunk 2 above)
Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/ei/services/SubscriptionService.java]: Fixed.

File: src/main/java/com/ericsson/ei/services/SubscriptionService.java (hunk 3 above)
Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/ei/services/SubscriptionService.java]: Done

File: src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java (hunk 4 above)
Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java]: Done.

**Pull request merged**